File: Frontend/src/types/message.ts
export type Chat = {
    id: string
    otherProfileImage: string
    otherUsername: string
    otherUserID: string
    lastMessage: Message
    createdAt: string
}

export interface Message {
    id: string
    chatId: string
    senderId: string
    content: string
    timestamp: string
    readed: boolean
}

File: Backend/models/message.go
package models

import (
	"time"
)

type Message struct {
	ID        string    `json:"id"`
	ChatID    string    `json:"chatId"`
	SenderID  string    `json:"senderId"`
	Content   string    `json:"content"`
	Readed    bool      `json:"readed"`
	Timestamp time.Time `json:"timestamp"`
}
